lib/ui/text/paragraph.cc

```diff
@@ -24,172 +26,91 @@ namespace blink {
 
 IMPLEMENT_WRAPPERTYPEINFO(ui, Paragraph);
 
-#define FOR_EACH_BINDING(V)         \
-  V(Paragraph, width)               \
-  V(Paragraph, height)              \
-  V(Paragraph, minIntrinsicWidth)   \
-  V(Paragraph, maxIntrinsicWidth)   \
-  V(Paragraph, alphabeticBaseline)  \
-  V(Paragraph, ideographicBaseline) \
-  V(Paragraph, didExceedMaxLines) \
-  V(Paragraph, layout)              \
-  V(Paragraph, paint)               \
-  V(Paragraph, getWordBoundary)     \
-  V(Paragraph, getRectsForRange)    \
-  V(Paragraph, getPositionForOffset)
+#define FOR_EACH_BINDING(V)          \
+  V(Paragraph, width)                \
+  V(Paragraph, height)               \
+  V(Paragraph, minIntrinsicWidth)    \
+  V(Paragraph, maxIntrinsicWidth)    \
+  V(Paragraph, alphabeticBaseline)   \
+  V(Paragraph, ideographicBaseline)  \
+  V(Paragraph, didExceedMaxLines)    \
+  V(Paragraph, layout)               \
+  V(Paragraph, paint)                \
+  V(Paragraph, getWordBoundary)      \
+  V(Paragraph, getRectsForRange)     \
+  V(Paragraph, getPositionForOffset) \
+  V(Paragraph, toggleTxt)
 
 DART_BIND_ALL(Paragraph, FOR_EACH_BINDING)
 
-Paragraph::Paragraph(PassOwnPtr<RenderView> renderView)
-    : m_renderView(renderView) {}
+Paragraph::Paragraph(PassOwnPtr<RenderView> renderView) {
+  m_paragraphImpl = std::make_unique<ParagraphImplBlink>(renderView);
+}
+
+Paragraph::Paragraph(const std::unique_ptr<txt::Paragraph>& paragraph) {
+  m_paragraphImpl = std::make_unique<ParagraphImplTxt>(
+      const_cast<std::unique_ptr<txt::Paragraph>&>(paragraph));
```

Comment:
This const_cast doesn't make any sense.  Presumably you want to take the |paragraph| argument as |std::unique_ptr<txt::Paragraph> paragraph| instead and use std::move to move the unique_ptr into the ParagraphImplTxt.